packages/openops/src/lib/openops-tables/rows.ts

```diff
@@ -63,96 +63,147 @@ const semaphore = TablesAccessSemaphore.getInstance();
 
 async function executeWithConcurrencyLimit<T>(
   fn: () => Promise<T>,
+  onError: (error: Error) => void,
 ): Promise<T> {
   const [value, release] = await semaphore.acquire();
   try {
     return await fn();
   } catch (error) {
-    logger.error('Error in locked row operation:', error);
+    onError(error as Error);
     throw error;
   } finally {
     release();
   }
 }
 
 export async function getRows(getRowsParams: GetRowsParams) {
-  return executeWithConcurrencyLimit(async () => {
-    if (
-      getRowsParams.filters &&
-      getRowsParams.filters.length > 1 &&
-      getRowsParams.filterType == null
-    ) {
-      throw new Error('Filter type must be provided when filters are provided');
-    }
+  if (
```

Comment:
took it outside of executeWithConcurrencyLimit, so that I will have the URL for the logs, and it doesn't need to be inside the lock anyway